packages/neon/neon_files/lib/sync/sources.dart

```diff
@@ -0,0 +1,161 @@
+part of '../neon_files.dart';
+
+class FilesSyncSources implements SyncSources<WebDavFile, FileSystemEntity> {
+  FilesSyncSources(
+    final NextcloudClient client,
+    final List<String> webdavBaseDir,
+    final String ioBaseDir,
+  )   : sourceA = FilesSyncSourceWebDavFile(client, webdavBaseDir),
+        sourceB = FilesSyncSourceFileSystemEntity(client, ioBaseDir);
+
+  @override
+  SyncSource<WebDavFile, FileSystemEntity> sourceA;
+
+  @override
+  SyncSource<FileSystemEntity, WebDavFile> sourceB;
+
+  @override
+  SyncConflictSolution? findSolution(final SyncObject<WebDavFile> objectA, final SyncObject<FileSystemEntity> objectB) {
+    if (objectA.data.isDirectory && objectB.data is Directory) {
+      return SyncConflictSolution.overwriteA;
+    }
+
+    return null;
+  }
+}
+
+class FilesSyncSourceWebDavFile implements SyncSource<WebDavFile, FileSystemEntity> {
+  FilesSyncSourceWebDavFile(
+    this.client,
+    this.webdavBaseDir,
+  );
+
+  /// [NextcloudClient] used by the WebDAV part.
+  final NextcloudClient client;
+
+  /// Base directory on the WebDAV server.
+  final List<String> webdavBaseDir;
+
+  final props = WebDavPropWithoutValues.fromBools(
+    davgetetag: true,
+    davgetlastmodified: true,
+    nchaspreview: true,
+    ocsize: true,
+    ocfavorite: true,
+  );
+
+  String _path(final SyncObject object) => [...webdavBaseDir, object.id].join('/');
```

Comment:
I hate all the path handling.

I was thinking to write a custom FilesUri class that implements uri and gives us these features (like removing trailing `/`, joining path + filename ...

What do you think? This is more a general question that would fit with #568

Reply:
Yeah, it's pretty ugly... A helper class or something like that would be pretty useful